core/src/main/java/com/linecorp/armeria/server/http/auth/BasicToken.java

```diff
@@ -0,0 +1,57 @@
+package com.linecorp.armeria.server.http.auth;
+
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The bearer token of <a href="https://en.wikipedia.org/wiki/Basic_access_authentication">HTTP Basic access authentication</a>.
+ * */
+public final class BasicToken {
+    private static final Pattern DECODED_PATTERN = Pattern.compile("(?<username>[^:]+):(?<password>\\S+)");
+    private static final Encoder BASE64_ENCODER = Base64.getEncoder();
+    private static final Decoder BASE64_DECODER = Base64.getDecoder();
+
+    private final String username;
+    private final String password;
+
+    private BasicToken(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
+
+    public String username() {
+        return username;
+    }
+
+    public String password() {
+        return password;
+    }
+
+    public String encode() {
+        return BASE64_ENCODER.encodeToString(String.format("%s:%s", username, password).getBytes());
```

Comment:
- `String.format()` is very slow. Use simple concat wherever possible. `username + ':' + password`
- Specify StandardCharsets.US_ASCII to getBytes()